### PR TITLE
DPE-2764 Add new extensions/plugins

### PR DESCRIPTION
**Pull request by TakoB222**
# Issue:

The list of supported plugins was too short: https://charmhub.io/postgresql/docs/r-supported-plugins-extensions

# Solution:

Add new extensions/plugins:

 * bloom
 * btree_gin
 * btree_gist
 * cube
 * dict_int
 * dict_xsyn
 * earthdistance
 * fuzzystrmatch
 * intarray
 * isn
 * lo
 * ltree
 * old_snapshot
 * pg_freespacemap
 * pgrowlocks
 * pgstattuple
 * pg_visibility
 * seg
 * tablefunc
 * tcn
 * tsm_system_rows
 * tsm_system_time
 * uuid-ossp

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -26,6 +26,98 @@ options:
...
+    description: Enable tcn extension
+  plugin_tsm_system_rows_enable:
+    default: false
+    type: boolean
+    description: Enable tsm_system_rows extension
+  plugin_tsm_system_time_enable:
+    default: false
+    type: boolean
+    description: Enable tsm_system_time extension
+  plugin_uuid-ossp_enable:
```
Comment (delgod) [inline config.yaml:117]:
```suggestion
  plugin_uuid_ossp_enable:
```

Comment (delgod) [reply config.yaml:117]: Unfortunately, the `-` symbol is prohibited for config options, please implement a workaround for extensions with `-` in the name.

**Review by delgod — APPROVED** [review]

File: tests/integration/test_plugins.py [hunk 2]
```diff
@@ -70,6 +99,98 @@ async def test_plugins(ops_test: OpsTest) -> None:
...
+        with pytest.raises(psycopg2.Error):
+            connection.cursor().execute(SEG_EXTENSION_STATEMENT)
+
+        # Test tablefunc extension disabled.
+        with pytest.raises(psycopg2.Error):
+            connection.cursor().execute(TABLEFUNC_EXTENSION_STATEMENT)
+
+        # Test tcn extension disabled.
+        with pytest.raises(psycopg2.Error):
+            connection.cursor().execute(TABLEFUNC_EXTENSION_STATEMENT)
```
Comment (taurus-forever) [inline tests/integration/test_plugins.py:181]:
```suggestion
            connection.cursor().execute(TCN_EXTENSION_STATEMENT)
```

**Review by taurus-forever — CHANGES_REQUESTED** [review]
Thank you! LGTM in general, but one typo.

@dragomirp please contact Artem in public MM and explain how to update library to pass CI tests. Thank you!

BTW: I have updated PR commit message to represent changes. TODO for Artem: 
1) add K8s PR
2) update documentation: 
* https://charmhub.io/postgresql/docs/r-supported-plugins-extensions
* https://charmhub.io/postgresql-k8s/docs/r-supported-plugins-extensions

**Review by marceloneppel — APPROVED** [review]

File: config.yaml [hunk 3]
```diff
@@ -26,6 +26,98 @@ options:
...
+    type: boolean
+    description: Enable tsm_system_rows extension
+  plugin_tsm_system_time_enable:
+    default: false
+    type: boolean
+    description: Enable tsm_system_time extension
+  plugin_uuid_ossp_enable:
+    default: false
+    type: boolean
+    description: Enable uuid_ossp extension
```
Comment (dragomirp) [inline config.yaml:120]:
```suggestion
    description: Enable uuid-ossp extension
```

File: src/charm.py [hunk 4]
```diff
@@ -927,6 +927,9 @@ def enable_disable_extensions(self, database: str = None) -> None:
 
             # Enable or disable the plugin/extension.
             extension = "_".join(plugin.split("_")[1:-1])
+            if extension == "uuid_ossp":
+                extension = '"uuid-ossp"'
+
```
Comment (dragomirp) [inline src/charm.py:932]: Like in k8s, I think it's better to have a single override mapping.

File: tests/integration/test_plugins.py
Comment (dragomirp) [inline tests/integration/test_plugins.py:1]: Like in k8s, I think it's better to have a single mapping and loop over it in config and test assertions.

**Comment by dragomirp** [comment]
Hi, @TakoB222, added a few nitpicks. Please resync with main, there are conflicting changes but also a fix for the check lib failure.

**Comment by dragomirp** [comment]
Hi, @TakoB222, please sync with main. This should handle any remaining CI breakages.

**Review by dragomirp — DISMISSED** [review]

**Review by marceloneppel — DISMISSED** [review]

**Review by marceloneppel — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**